Source/GameMapEx.cpp

```diff
@@ -115,28 +115,9 @@ int GameMapEx::LoadMap(char *fileName)
 	return func(p, 0, fileName);
 }
 
-void GameMapEx::CopyTileMap(int* tileMap, int xMin, int xMax, int yMin, int yMax)
+void GameMapEx::CopyTileMap(int* tileMap)
 {
-	OP2Map *p = (OP2Map*)mapObj;
 	int **tileArray = (int**)(mapTileData);
-
-	int mapWidth = xMax - xMin;
-
-	for (int x = xMin; x < xMax; ++x)
-	{
-		for (int y = yMin; y < yMax; ++y)
-		{
-			int px = x - xMin;
-			int py = y - yMin;
-
-			int xLower = p->tileXMask & x;
-			int xUpper = xLower >> 5;
-			xLower &= 31;
-			xUpper <<= p->logTileHeight;
-			xUpper += y;
-			xUpper <<= 5;
-			
-			tileMap[px + py * mapWidth] = (*tileArray)[xUpper + xLower];
-		}
-	}
+	
+	memcpy(tileMap, *tileArray, GetMapWidth()*GetMapHeight()*sizeof(int));

```

Comment:
The second parameter should be offset for non-around-the-world maps. As the copy size is bounded to the valid data size, you'll want to skip over the padding to ensure you return all of the correct data.

In the following, the compiler will automatically handle the `* sizeof(TileData)` as part of the pointer offset arithmetic. C++ is weird that way, with how addition to pointers is auto scaled by the element size.
```cpp
std::size_t offset = GetMapWidth() >= 512 ? 0 : 32 * height;
memcpy(tileMap, *tileArray + offset, GetMapWidth()*GetMapHeight()*sizeof(int));
```

Reply:
GetMapWidth() is returning the internal double width. For what you are describing, it would have to be divided by two?
Actually, skipping the padding is a good optimization.

Reply:
You would be correct then.

I kind of feel like that function should not be returning the internal padding width though. It would seem to make more sense if it returned the user visible width. Maybe we should discuss that in a separate issue.

Reply:
I'm going to check clipRect to see if I can detect the visible area so I can remove padding. Removing padding will affect how this function is used.

Reply:
If padding is removed by the copy, the only relevant attributes are then `width` and `height`. In a way, this simplifies how the result is used.